Modify the guest session script so that it doesn't depend on the Debian/
Ubuntu specific adduser package.
--- a/debian/guest-account
+++ b/debian/guest-account
@@ -10,7 +10,12 @@
 
 add_account ()
 {
-  HOME=`mktemp -td guest-XXXXXX`
+  NEWHOME=`mktemp -td guest-XXXXXX`
+  # useradd does not accept upper case characters in the username
+  HOME=`echo $NEWHOME | tr '[A-Z]' '[a-z]'`
+  if [ "x$NEWHOME" != "x$HOME" ]; then
+    mv $NEWHOME $HOME
+  fi
   USER=`echo $HOME | sed 's/\(.*\)guest/guest/'`
 
   # if $USER already exists, it must be a locked system account with no existing
@@ -36,7 +41,21 @@
     fi
   else
     # does not exist, so create it
-    adduser --system --no-create-home --home / --gecos "Guest" --group --shell /bin/bash $USER || {
+    # Arch Linux modification: Ubuntu/Debian uses their own adduser package,
+    # which works differently from the own provided by the shadow package.
+    # Instead, use useradd, which works in any distro.
+    #
+    # Only the syntax is changed
+    # adduser          -> useradd
+    # --system         -> --system
+    # --no-create-home -> --no-create-home
+    # --home           -> --home-dir
+    # --gecos          -> --comment
+    # --group          -> --user-group
+    # --shell          -> --shell
+    #
+    #adduser --system --no-create-home --home / --gecos "Guest" --group --shell /bin/bash $USER || {
+    useradd --system --no-create-home --home-dir / --comment "Guest" --user-group --shell /bin/bash $USER || {
         umount "$HOME"
         rm -rf "$HOME"
         exit 1
@@ -109,10 +128,15 @@
     echo "Error: invalid user $USER"
     exit 1
   }
-  UID=`echo "$PWENT" | cut -f3 -d:`
+  USERUID=`echo "$PWENT" | cut -f3 -d:`
   HOME=`echo "$PWENT" | cut -f6 -d:`
 
-  if [ "$UID" -ge 500 ]; then
+  # deluser is provided by the adduser package on Debian/Ubuntu. useradd
+  # doesn't have a '--system' parameter, which causes deluser to only delete
+  # system users, so this will be handled using this script.
+  SYS_UID_MIN="$(cat /etc/login.defs | grep SYS_UID_MIN | awk '{print $2}')"
+  SYS_UID_MAX="$(cat /etc/login.defs | grep SYS_UID_MAX | awk '{print $2}')"
+  if [ "$USERUID" -lt "$SYS_UID_MIN" ] || [ "$USERUID" -gt "$SYS_UID_MAX" ]; then
     echo "Error: user $USER is not a system user."
     exit 1
   fi
@@ -132,9 +156,10 @@
   rm -rf "$HOME"
 
   # remove leftovers in /tmp
-  find /tmp -mindepth 1 -maxdepth 1 -uid "$UID" -print0 | xargs -0 rm -rf || true
+  find /tmp -mindepth 1 -maxdepth 1 -uid "$USERUID" -print0 | xargs -0 rm -rf || true
 
-  deluser --system "$USER"
+  #deluser --system "$USER"
+  userdel "$USER"
 }
 
 case "$1" in
